src/ferny/session.py

```diff
@@ -93,8 +91,9 @@ async def connect(self,
                       pkcs11: Optional[str] = None,
                       port: Optional[int] = None,
                       interaction_responder: Optional[InteractionHandler] = None) -> None:
-        os.makedirs(FERNY_DIR, exist_ok=True)
-        self._controldir = tempfile.TemporaryDirectory(dir=FERNY_DIR)
+        rundir = os.path.join(os.environ.get('XDG_RUNTIME_DIR', '/run'), 'ferny')
```

Comment:
This feels like a bad fallback -- users can't write into /run/. This should probably fall back to mkdtemp()?

Reply:
I think the idea here was that running as a user requires `XDG_RUNTIME_DIR` to be set, and if it's not set, we assume it's root and use `/run`.  We could make this more explicit by checking the uid, but the issue you cite here is preexisting and I hesitate to change this around more than we need to — Session is on its way to less and less use, after all...

Reply:
Ah, I see now that it's moved code. Okay.